memory/libhugetlbfs.py

```diff
@@ -98,11 +100,37 @@ def setUp(self):
         os.chdir(self.srcdir)
         patch = self.params.get('patch', default='elflink.patch')
         process.run('patch -p1 < %s' % data_dir + '/' + patch, shell=True)
+
+        if (detected_distro.name == "redhat" or
+                detected_distro.name == "fedora"):
+            falloc_patch = 'patch -p1 < %s ' % (
+                os.path.join(data_dir, 'falloc.patch'))
+            process.run(falloc_patch, shell=True)
+
         build.make(self.srcdir, extra_args='BUILDTYPE=NATIVEONLY')
 
+    def fail_check(self, log, check):
```

Comment:
Yep. Not tested code, provided only as reference. Thanks.
You can also include a `continue` just bellow the `section = True` and the `section = False`. This will improve a little bit the performance.